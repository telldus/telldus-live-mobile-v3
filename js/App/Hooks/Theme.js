/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
// @flow
'use strict';
import React from 'react';
import {
	useColorScheme,
} from 'react-native';
import {
	useSelector,
} from 'react-redux';
import { useIntl } from 'react-intl';
import { CONSTANTS } from 'live-shared-data';
const {
	DEVICE_THEME_KEY,
	DARK_THEME_KEY,
} = CONSTANTS;

import ThemedColorsOne from '../Theme/ThemedColors';
import ThemedColorsTwo from '../Theme/ThemedColorsTwo';
import i18n from '../Translations/common';

import {
	getThemeSetOptions,
} from '../Lib/appUtils';

const getThemedColors = (selectedThemeSet: Object): Object => {
	switch (selectedThemeSet.key) {
		case 1:
			return ThemedColorsOne;
		case 2:
			return ThemedColorsTwo;
		default:
			return ThemedColorsTwo;
	}
};

const getThemeData = (themeInApp: string | null = 'light', ThemedColors: Object): Object => {
	const colors = ThemedColors[themeInApp] || ThemedColors.light;
	return {colors};
};

const useAppTheme = (): Object => {
	const colorScheme = useColorScheme();
	const { defaultSettings = {} } = useSelector((state: Object): Object => state.app);
	const {
		themeInApp = DEVICE_THEME_KEY,
		selectedThemeSet,
	} = defaultSettings;
	const options = getThemeSetOptions();
	const _selectedThemeSet = (selectedThemeSet && selectedThemeSet.key) ? selectedThemeSet : options[1];
	return React.useMemo((): Object => {
		const ThemedColors = getThemedColors(_selectedThemeSet);
		if (themeInApp === DEVICE_THEME_KEY) {
			return {
				colorScheme,
				dark: colorScheme === DARK_THEME_KEY,
				themeInApp,
				selectedThemeSet: _selectedThemeSet,
				...getThemeData(colorScheme, ThemedColors),
			};
		}
		return {
			colorScheme,
			dark: themeInApp === DARK_THEME_KEY,
			themeInApp,
			selectedThemeSet: _selectedThemeSet,
			...getThemeData(themeInApp, ThemedColors),
		};
	}, [
		colorScheme,
		themeInApp,
		_selectedThemeSet,
	]);
};

const useAppThemeOptions = (): Object => {
	const colorScheme = useColorScheme();

	const { defaultSettings = {} } = useSelector((state: Object): Object => state.app);
	const {
		selectedThemeSet,
	} = defaultSettings;
	const optionsTS = getThemeSetOptions();
	const _selectedThemeSet = (selectedThemeSet && selectedThemeSet.key) ? selectedThemeSet : optionsTS[0];
	const ThemedColors = getThemedColors(_selectedThemeSet);
	const themes = Object.keys(ThemedColors);

	const intl = useIntl();
	const {
		formatMessage,
	} = intl;

	let options = themes.map((value: string): Object => {
		let label = formatMessage(i18n.themeOption1);
		if (value === 'dark') {
			label = formatMessage(i18n.themeOption2);
		}
		return {
			value,
			label,
		};
	});
	options.push({
		value: DEVICE_THEME_KEY,
		label: formatMessage(i18n.themeOption3),
	});
	return {
		options,
		colorScheme,
	};
};

module.exports = {
	useAppTheme,
	useAppThemeOptions,
};
