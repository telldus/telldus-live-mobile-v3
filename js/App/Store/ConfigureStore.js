/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from './Promise';
import array from './Array';
import analytics from './Analytics';
import reducers from 'Reducers';
import createLogger from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import { AsyncStorage } from 'react-native';

let isDebuggingInChrome = __DEV__ && !!window.navigator.userAgent;

let logger = createLogger({
	predicate: (getState, action) => isDebuggingInChrome,
	collapsed: true,
	duration: true,
});

let createTheStore = applyMiddleware(thunk, promise, array, analytics, logger)(createStore);

let _store;
export function configureStore(onComplete: ?() => void) {
	const store = autoRehydrate()(createTheStore)(reducers);
	persistStore(store, {storage: AsyncStorage}, onComplete);
	if (isDebuggingInChrome) {
		window.store = store;
	}
	_store = store; // TODO: fix this ugly stuff
	return store;
}
export function getStore() { return _store; }
