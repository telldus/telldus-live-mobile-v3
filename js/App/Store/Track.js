/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import type { Action } from '../actions/types';
import DeviceInfo from 'react-native-device-info';

import { Analytics, Hits as GAHits } from 'react-native-google-analytics';
import { googleAnalyticsId } from 'Config';

function track(action: Action): void {

	let clientId = DeviceInfo.getUniqueID();
	let ga = new Analytics(googleAnalyticsId, clientId, 1, DeviceInfo.getUserAgent());

	if (action.type === 'SWITCH_TAB') {
		let screenView = new GAHits.ScreenView(
			'Telldus Live! app',
			action.tab,
			DeviceInfo.getVersion(),
			DeviceInfo.getBundleId()
		);
		ga.send(screenView);
	} else {
		let gaEvent = new GAHits.Event('Action', action.type);
		ga.send(gaEvent);
	}
}

module.exports = track;
