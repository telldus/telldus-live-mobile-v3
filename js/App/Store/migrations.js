/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import {
	DEFAULT_DASHBOARD_ID,
} from '../Lib/dashboardUtils';

export default function migrations(state: Object = {}): Promise<any> {
	const { tabs, ...withOutTabs } = state;

	let newState = {...state};
	// Remove redux store object 'tabs' if present.
	if (tabs) {
		newState = withOutTabs;
	}

	const { App, ...withOutApp } = newState;
	// Rename redux store object 'App' to 'app'.
	if (App) {
		newState = {
			...withOutApp,
			app: App,
		};
	}

	const { app, dashboard, user } = newState;
	if (app && !app.defaultSettings) {
		newState = {
			...newState,
			app: {
				...app,
				defaultSettings: {
					dimmerSensitivity: 5,
					sortingDB: 'Chronological',
					activeDashboardId: DEFAULT_DASHBOARD_ID,
				},
			},
		};
	}

	if (app && app.defaultSettings && !app.defaultSettings.activeDashboardId) {
		newState = {
			...newState,
			app: {
				...app,
				defaultSettings: {
					...app.defaultSettings,
					activeDashboardId: DEFAULT_DASHBOARD_ID,
				},
			},
		};
	}

	if (user && !user.activeDashboardId) {
		newState = {
			...newState,
			user: {
				...user,
				activeDashboardId: DEFAULT_DASHBOARD_ID,
			},
		};
	}

	let { userId, userProfile = {} } = user || {};
	if (!userId) {
		const {
			uuid,
		} = userProfile;
		userId = uuid;
	}

	if (dashboard && userId) {
		const {
			devicesById = {},
			sensorsById = {},
			sensorIds,
			deviceIds,
		} = dashboard;

		const prevDataType = [true, false];

		let newDashboard = {...dashboard};

		const devicesByIdC = Object.keys(devicesById);
		if (devicesByIdC.length > 0 && prevDataType.indexOf(devicesById[devicesByIdC[0]]) !== -1) {
			newDashboard = {
				...newDashboard,
				devicesById: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: devicesById,
					},
				},
				deviceIds: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: deviceIds,
					},
				},
			};
		}
		const sensorsByIdC = Object.keys(sensorsById);
		if (sensorsByIdC.length > 0 && prevDataType.indexOf(sensorsById[sensorsByIdC[0]]) !== -1) {
			newDashboard = {
				...newDashboard,
				sensorsById: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: sensorsById,
					},
				},
				sensorIds: {
					[userId]: {
						[DEFAULT_DASHBOARD_ID]: sensorIds,
					},
				},
			};
		}
		newState = {
			...newState,
			dashboard: {
				...newDashboard,
			},
		};
	}

	const { gateways, sensorsList } = newState;
	if (gateways) {
		// Insert the attribute/property 'localKey' to each gateways object if not already present.
		const newGateways = insertLocalKeyAttribute(gateways);
		newState = {
			...newState,
			gateways: newGateways,
		};
	}

	if (sensorsList && sensorsList.defaultTypeById) {
		const defaultSensorSettings = migrateToDefaultSensorSettings(sensorsList.defaultTypeById);
		newState = {
			...newState,
			sensorsList: {
				defaultSensorSettings,
			},
		};
	}

	if (sensorsList && sensorsList.defaultSensorSettings) {
		let newDefaultSensorSettings = {};
		let { defaultSensorSettings } = sensorsList;
		for (let key in defaultSensorSettings) {
			const { historySettings = {}, ...others } = defaultSensorSettings[key];
			const { selectedTwo, selectedOne, ...othersH } = historySettings;
			if ((selectedTwo && selectedTwo.type === 'wdir') || (selectedOne && selectedOne.type === 'wdir')) {
				newDefaultSensorSettings[key] = {
					...others,
					historySettings: {
						...othersH,
						selectedTwo: undefined,
						selectedOne: undefined,
					},
				};
			} else {
				newDefaultSensorSettings[key] = defaultSensorSettings[key];
			}
		}

		newState = {
			...newState,
			sensorsList: {
				defaultSensorSettings: newDefaultSensorSettings,
			},
		};
	}

	return Promise.resolve(newState);
}


const insertLocalKeyAttribute = (gateways: Object): Object => {
	const { byId } = gateways;
	let newById = {};
	for (let key in byId) {
		let gateway = byId[key];
		if (gateway && !gateway.localKey) {
			gateway = {
				...gateway,
				localKey: {
					key: null,
					ttl: null,
					uuid: null,
					address: null,
					port: null,
					macAddress: null,
					supportLocal: false,
				},
			};
		}
		newById[key] = gateway;
	}
	return {
		...gateways,
		byId: newById,
	};
};


const migrateToDefaultSensorSettings = (defaultTypeById: Object): Object => {
	let defaultSensorSettings = {};
	for (let key in defaultTypeById) {
		defaultSensorSettings[key] = {
			displayType: defaultTypeById[key],
			displayTypeDB: null,
			historySettings: {},
		};
	}
	return defaultSensorSettings;
};
