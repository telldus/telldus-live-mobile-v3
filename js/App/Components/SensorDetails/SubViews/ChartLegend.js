/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.
 * org/licenses/>.
 */

// @flow

'use strict';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {
	View,
	Text,
	IconTelldus,
	RippleButton,
} from '../../../../BaseComponents';

import shouldUpdate from '../../../Lib/shouldUpdate';

import Theme from '../../../Theme';

type Props = {
	legendData: Array<any>,
	appLayout: Object,
	onPressToggleView: () => void,
	onPressResetChartView: () => void,
	fullscreen: boolean,
};

export default class ChartLegend extends View<Props, null> {
	props: Props;
	constructor(props: Props) {
		super(props);
	}

	shouldComponentUpdate(nextProps: Object, nextState: Object): boolean {
		const propsChange = shouldUpdate(this.props, nextProps, [
			'legendData', 'appLayout', 'fullscreen',
		]);
		if (propsChange) {
			return propsChange;
		}
		return false;
	}

	render(): Object | null {
		const { legendData, appLayout, onPressToggleView, onPressResetChartView, fullscreen } = this.props;
		const { rowTextColor } = Theme.Core;
		const {
			containerStyle,
			labelContainerStyle,
			iconStyle,
			labelStyle,
			fontSizeFullscreenIcon,
			legendsContainerStyle,
			btnContainerIconStyle,
		} = this.getStyles(appLayout);

		return (
			<View style={containerStyle}>
				<View style={legendsContainerStyle}>
					{legendData.map((item: Object, index: number): Object | null => {
						const { icon, onPress, value, color } = item;
						if (!value) {
							return null;
						}
						return (
							<RippleButton
								key={index}
								style={labelContainerStyle}
								onPress={onPress}>
								<IconTelldus icon={icon} style={{ ...iconStyle, color }} />
								<Text style={[labelStyle, { color }]} numberOfLines={1}>
									{value}
								</Text>
							</RippleButton>
						);
					})
					}
				</View>
				<View style={[btnContainerIconStyle]}>
					<TouchableOpacity onPress={onPressToggleView}>
						<Icon name={fullscreen ? 'fullscreen-exit' : 'fullscreen'} size={fontSizeFullscreenIcon} color={rowTextColor} />
					</TouchableOpacity>
					{!fullscreen &&
						<TouchableOpacity onPress={onPressResetChartView}>
							<IconTelldus icon={'resetzoom'} style={{ fontSize: fontSizeFullscreenIcon - 4, color: rowTextColor }} />
						</TouchableOpacity>
					}
				</View>
			</View>
		);
	}

	getStyles(appLayout: Object): Object {
		const { height, width } = appLayout;
		const isPortrait = height > width;
		const deviceWidth = isPortrait ? width : height;

		const fontSizeLabel = deviceWidth * 0.038;
		const fontSizeIcon = deviceWidth * 0.06;
		const fontSizeFullscreenIcon = deviceWidth * 0.07;

		return {
			containerStyle: {
				flex: 0,
				flexDirection: 'row',
				marginTop: 20,
				marginLeft: 10,
				paddingBottom: 20,
			},
			legendsContainerStyle: {
				flex: 1,
				flexDirection: 'row',
			},
			labelContainerStyle: {
				flex: 1,
				flexDirection: 'row',
				justifyContent: 'flex-start',
				alignItems: 'center',
				marginRight: 10,
			},
			iconStyle: {
				fontSize: fontSizeIcon,
				marginRight: 5,
			},
			labelStyle: {
				flex: 1,
				fontSize: fontSizeLabel,
			},
			btnContainerIconStyle: {
				flex: 0,
				padding: 0,
				width: 30,
				height: 40,
				alignItems: 'center',
				justifyContent: 'center',
			},
			fontSizeFullscreenIcon,
		};
	}
}
