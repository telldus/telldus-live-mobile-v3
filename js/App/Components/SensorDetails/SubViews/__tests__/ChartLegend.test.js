/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */


import React from 'react';
import {
	Dimensions,
} from 'react-native';
import { configureStore } from '../../../../Store/ConfigureStore';
import {act} from 'react-test-renderer';

import {
	rendererWithIntlAndReduxProviders,
} from '../../../../../Utils/jestUtils';

import ChartLegend from '../ChartLegend';
import {
	setAppLayout,
} from '../../../../Actions';
import Theme from '../../../../Theme';

let {height, width} = Dimensions.get('window');

const store = configureStore().store;

jest.useFakeTimers();


const onOp = () => {
};

const selectedOne = {
	type: 'selectedOneTpe',
	scale: 'selectedOneScale',
	value: 'selectedOneValue',
	icon: 'selectedOneIcon',
};

const selectedTwo = {
	type: 'selectedTwoTpe',
	scale: 'selectedTwoScale',
	value: 'selectedTwoValue',
	icon: 'selectedTwoIcon',
};

const { brandDanger, brandInfo } = Theme.Core;

const colorsScatter = [brandDanger, brandInfo];

describe('<ChartLegend /> - snapshot', () => {
	it('renders ChartLegend when chart not fullscreen, both scales shown', () => {
		let component;
		const showOne = true;
		const showTwo = true;
		const legendData = [{
			...selectedOne,
			onPress: onOp,
			color: showOne ? colorsScatter[0] : Theme.Core.inactiveTintColor,
		},
		{
			...selectedTwo,
			onPress: onOp,
			color: showTwo ? colorsScatter[1] : Theme.Core.inactiveTintColor,
		}];
		act(() => {
			store.dispatch(setAppLayout({
				height,
				width,
			}));

			component = rendererWithIntlAndReduxProviders(
				<ChartLegend
					appLayout={{
						height,
						width,
					}}
					legendData={legendData}
					fullscreen={false}
					onPressResetChartView={onOp}
					onPressToggleView={onOp}/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('renders ChartLegend when chart fullscreen, scale one shown and scale two not shown', () => {
		let component;
		const showOne = true;
		const showTwo = false;
		const legendData = [{
			...selectedOne,
			onPress: onOp,
			color: showOne ? colorsScatter[0] : Theme.Core.inactiveTintColor,
		},
		{
			...selectedTwo,
			onPress: onOp,
			color: showTwo ? colorsScatter[1] : Theme.Core.inactiveTintColor,
		}];
		act(() => {
			store.dispatch(setAppLayout({
				height,
				width,
			}));

			component = rendererWithIntlAndReduxProviders(
				<ChartLegend
					appLayout={{
						height,
						width,
					}}
					legendData={legendData}
					fullscreen={true}
					onPressResetChartView={onOp}
					onPressToggleView={onOp}/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
