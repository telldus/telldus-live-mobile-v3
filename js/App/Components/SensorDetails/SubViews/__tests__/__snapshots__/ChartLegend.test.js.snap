// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartLegend /> - snapshot renders ChartLegend when chart fullscreen, scale one shown and scale two not shown 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 0,
      "flexDirection": "row",
      "marginLeft": 10,
      "marginTop": 20,
      "paddingBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        importantForAccessibility="yes"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#d9534f",
            "fontFamily": "telldusicons",
            "fontSize": 45,
            "marginRight": 5,
          }
        }
      >
        selectedOneIcon
      </Text>
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": undefined,
              "flex": 1,
              "fontFamily": "SuisseIntl",
              "fontSize": 28.5,
            },
            Object {
              "color": "#d9534f",
            },
          ]
        }
      >
        selectedOneValue
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 0,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        importantForAccessibility="yes"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#A59F9A",
            "fontFamily": "telldusicons",
            "fontSize": 45,
            "marginRight": 5,
          }
        }
      >
        selectedTwoIcon
      </Text>
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": undefined,
              "flex": 1,
              "fontFamily": "SuisseIntl",
              "fontSize": 28.5,
            },
            Object {
              "color": "#A59F9A",
            },
          ]
        }
      >
        selectedTwoValue
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 0,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 0,
          "height": 40,
          "justifyContent": "center",
          "padding": 0,
          "width": 30,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#8e8e93",
              "fontSize": 52.50000000000001,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ChartLegend /> - snapshot renders ChartLegend when chart not fullscreen, both scales shown 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 0,
      "flexDirection": "row",
      "marginLeft": 10,
      "marginTop": 20,
      "paddingBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        importantForAccessibility="yes"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#d9534f",
            "fontFamily": "telldusicons",
            "fontSize": 45,
            "marginRight": 5,
          }
        }
      >
        selectedOneIcon
      </Text>
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": undefined,
              "flex": 1,
              "fontFamily": "SuisseIntl",
              "fontSize": 28.5,
            },
            Object {
              "color": "#d9534f",
            },
          ]
        }
      >
        selectedOneValue
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 0,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        importantForAccessibility="yes"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#5bc0de",
            "fontFamily": "telldusicons",
            "fontSize": 45,
            "marginRight": 5,
          }
        }
      >
        selectedTwoIcon
      </Text>
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": undefined,
              "flex": 1,
              "fontFamily": "SuisseIntl",
              "fontSize": 28.5,
            },
            Object {
              "color": "#5bc0de",
            },
          ]
        }
      >
        selectedTwoValue
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "borderRadius": 0,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 0,
          "height": 40,
          "justifyContent": "center",
          "padding": 0,
          "width": 30,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#8e8e93",
              "fontSize": 52.50000000000001,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        importantForAccessibility="yes"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#8e8e93",
            "fontFamily": "telldusicons",
            "fontSize": 48.50000000000001,
          }
        }
      >
        resetzoom
      </Text>
    </View>
  </View>
</View>
`;
